apps/extension/server/src/server.ts

```diff
@@ -192,21 +200,42 @@ async function handleDocumentChange(textDocument: TextDocument): Promise<void> {
     }
     const [settings, machineResults] = await Promise.all([
       getDocumentSettings(textDocument.uri),
-      filterOutIgnoredMachines(extracted).machines,
+      filterOutIgnoredMachines(extracted).machines.filter(isMachineResult),
```

Comment:
`filterOutIgnoredMachines` is already filtering this:
https://github.com/statelyai/xstate-tools/blob/a846c768072524f86a51a20b379480e932f768fb/packages/shared/src/filterOutIgnoredMachines.ts#L15-L17 
```suggestion
      filterOutIgnoredMachines(extracted).machines,
```

Reply:
True, but it's defined as possibly being undefined now with the revert of the type of machine in the `FileExtractResult` interface. So maybe I should make `filterOutIgnoredMachines` return a new type where the machines can't be undefined?

Reply:
I've tried to create a generically typed version of `FileExtractResult` in the latest commits, maybe that's better?